src/views/Pools/components/PoolCard/CardActions/index.tsx

```diff
@@ -0,0 +1,81 @@
+import BigNumber from 'bignumber.js'
+import React from 'react'
+import styled from 'styled-components'
+import { Flex, Text, Box } from '@pancakeswap-libs/uikit'
+import useI18n from 'hooks/useI18n'
+import { PoolCategory } from 'config/constants/types'
+import { Pool } from 'state/types'
+import ApprovalAction from './ApprovalAction'
+import StakeActions from './StakeActions'
+import HarvestActions from './HarvestActions'
+
+const InlineText = styled(Text)`
+  display: inline;
+`
+
+const CardActions: React.FC<{
+  pool: Pool
+  stakedBalance: BigNumber
+  accountHasStakedBalance: boolean
+  stakingTokenPrice: number
+}> = ({ pool, stakedBalance, accountHasStakedBalance, stakingTokenPrice }) => {
+  const { sousId, stakingToken, earningToken, harvest, poolCategory, userData } = pool
+  // Pools using native BNB behave differently than pools using a token
+  const isBnbPool = poolCategory === PoolCategory.BINANCE
+  const TranslateString = useI18n()
+  const allowance = new BigNumber(userData?.allowance || 0)
+  const stakingTokenBalance = new BigNumber(userData?.stakingTokenBalance || 0)
+  const earnings = new BigNumber(userData?.pendingReward || 0)
+  const needsApproval = !accountHasStakedBalance && !allowance.toNumber() && !isBnbPool
+  const isStaked = stakedBalance.toNumber() > 0
```

Comment:
same feedback, stick with BigNumber

Reply:
Akh sorry you had to catch this multiple times, ty! I wasn't familiar with `.gt()` / .`gte()` until yesterday. Suuuper useful